Kustannussuunnitelman osittaminen ym.

K1ll3rF0x opened the pull request:
Hox: Aiempiin committeihin liittyviä committeja vanhimmassa PR:ssä #2035
Tätä edeltävä PR: #2086

K1ll3rF0x committed:
Poista vanhaa koodia.

RelNotes:
Related:

K1ll3rF0x committed:
Erota hankintakustannukset osio omaan tiedostoonsa. Luo tiedosto
kustannussuunnitelma-näkymän yhteisille apufunktioille.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää kattohinnan tarkastus e2e-testeihin.

RelNotes:
Related:

K1ll3rF0x committed:
Erota erillishankinnat osio omaan nimiavaruuteensa.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Erota johto- ja hallintokorvaukset osio omaan nimiavaruuteensa.
* HOX: Tämän osion tallennuksessa näkyy olevan ongelmia. Oli myös ennen
nimiavaruuteen erottamista. Lisäksi sivun pääyhteenvetonäkymässä ei näy
osion summia. Summat ovat nollina joka hoitovuodelle vaikka osiossa selvästi
näkyy hoitovuosille summia ja indeksikorjattuja summia.

RelNotes:
Related:

K1ll3rF0x committed:
Whitespace muutoksia.
* Johto- ja hallintokorvaus osion grid-määrityksien leveyttä
kapeammaksi.

RelNotes:
Related:

K1ll3rF0x committed:
Erota hoidonjohtopalkkio osio omaan nimiavaruuteensa.

RelNotes:
Related:

K1ll3rF0x committed:
Erota tavoite- ja kattohinta osio omaan nimiavaruuteensa.

RelNotes:
Related:

K1ll3rF0x committed:
Erota tilaajan varaukset osio omaan nimiavaruuteensa.

RelNotes:
Related:

K1ll3rF0x committed:
Siisti kustannussuunnitelman päänäkymän koodia.

RelNotes:
Related:

K1ll3rF0x committed:
Nosta paikallinen cypress versioon 7.x

RelNotes:
Related:

K1ll3rF0x committed:
Nosta 'Lataa sivu'-testin timeouttia 10s -> 40s

RelNotes:
Related:

K1ll3rF0x committed:
Erota maarataulukon grid-apurit omaan tiedostoon.

RelNotes:
Related:

K1ll3rF0x committed:
Siirrä määrätaulukko-muotoiset grid-määritykset relevantteihin
nimiavaruuksiin.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typo suunnittellut->suunnitellut.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typoja koodissa piillota -> piilota.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typoja koodista: piillotettu -> piilotettu

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typoja koodista piillotetaan -> piilotetaan.

RelNotes:
Related:

K1ll3rF0x committed:
Ryhmittele kustannussuunitelman kontrollerin koodia paremmaksi, tee
whitespace korjauksia.

RelNotes:
Related:

K1ll3rF0x committed:
Selkeytä ja ryhmittele kustannussuunnitelman kontrolleria. Tee
whitespace korjauksia.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa kustannusuunnitelman yhteenvetonäkymään Johto- ja
hallintokorvaukset osion summat.
* Toimistkulut oli jäänyt lisäämättä yhteenvetonäkymän summaan.
* Tehty apufunktio, jota käytetään summan laskemisessa myös itse Johto-
ja hallintokorvaukset osiossa. Näin vähennetään riskiä, että vastaava
tilanne toistuu tämän osion kohdalla uudestaan.

[VHAR-5162]

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa funktion nimi.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa Johto- ja hallintokorvaus osion palkkojen tallentamisongelma.
* Jos käyttäjä täytti ensin taulukkoon tunnit ja sitten vasta palkan,
niin backend heitti nullpointer virhettä. Nyt varmistetaan, että
tuntipalkkaa ei käytetä laskennassa, ellei sitä ole määritelty.

[VHAR-5162]

RelNotes:
Related:

K1ll3rF0x committed:
Varmista, että kustannussuunnitelman cypress-testeissä tallennuskyselyt
menevät läpi ennen seuraavaan testiin siirtymistä.
* Tämän pitäisi korjata mahdolliset flakyt-testit lopullisesti.

RelNotes:
Related:

K1ll3rF0x committed:
Suorita tietokannan alustaminen omassa testiblokissaan, jotta se näkyy
selvemmin cypress-lokissa.

RelNotes:
Related:

K1ll3rF0x committed:
Tee blur-eventtien triggeröitymisestä robustimpaa kustannussuunnitelman
cypress-testeissä.

RelNotes:
Related:

K1ll3rF0x committed:
Ryhmittele kustannussuunnitelman e2e-testejä uusiksi.
* Testit kohdistetaan selkeästi osioon kerrallaan.
* Yleiset näkymän testit ja tietokannan alustus tehdään kerran aluksi.

RelNotes:
Related:

K1ll3rF0x committed:
Käytä samaan nimiavaruuteen viittavia qualified keywordeja kaikissa
kustannussuunnitelmaviewin gridien osioiden nimissä ja osioihin
viittauksissa.

[fixes VHAR-5193]

RelNotes:
Related:

K1ll3rF0x committed:
Heitä virhe gridin osien etsinnän yhteydessä, mikäli gridin osaa ei
löytynyt annetulla tunnisteella.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää data-cy attribuutti.

RelNotes:
Related:

K1ll3rF0x committed:
Siirrä tayta-alas-napin-toiminto hankintakustannukset osion tiedostoon.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa bugi, jossa toimenpiteen rahavarausten kopiointi allaoleviin
kuukausiin ei päivittänyt yhteenvetoja.
* Rahavarausten tayta-alas-napin-toiminto-funktion
"paivita-solun-arvo"-kutsuun piti lisätä ":triggeroi-seuranta? true",
jotta taulukko ja yhteenveto päivitetään "Kopioi allaoleviin"-napin
painamisen jälkeen.
* Rahavarausten-gridin grid-kasittelijan jarjestys-optioista oli jäänyt
metadata: ^{:nimi :mapit} määrittelemättä. Gridissä viitattiin tähän
:mapit-nimiseen järjestykseen, joka sotki lopulta datan väärään
:järjestykseen kun "tayta-alas-napin-toiminto"-funktiosta oli aktivoitu
":triggeroi-seuranta true"

RelNotes:
Related:

K1ll3rF0x committed:
Lisää kommentteja.

RelNotes:
Related:

K1ll3rF0x committed:
E2E-testaa koko hankintakustannukset osio.
* Lisätty testaukseen apufunktioita.
* Refaktoroitu kustannussuunnitelman E2E-testejä.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Korjaa vieritys-komponentin HTML id-attribuutteja.
* Kustannussuunnitelma-sivulla tuli helposti HTML ID-attribuuttien
kanssa collisioneita. Lisättiin majakan vieritys spaneille
loppuliitteeksi "-majakka", jotta riski ID-collisioneille pienenee.
* Muutos on sisäinen vierityskomponentille, eikä vaadi muutoksia muualle
näkymään.

RelNotes:
Related:

K1ll3rF0x committed:
Generoi uniikki elementin id yleis-suodattimen käyttämälle
pudotusvalikolle.
* Korjaa valinnan räpsymisen mikäli näkymässä on samanaikaisesti useita
yleis-suodatin komponentteja renderöitynä.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää ja muuta HTLM-attribuutteja cypress-testejä varten.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testit Erillishankinnat osiolle.
* Testejä kirjoittaessa löyty ongelmallinen UI-logiikka: VHAR-5213, jota
voisi bugiksin luonnehtia.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testit Hoidonjohtopalkkio osiolle.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testit Tilaajan rahavaraukset osiolle.

RelNotes:
Related:

K1ll3rF0x committed:
Siirrä kustannussuunnitelman Cypress apufunktiot eri tiedostoon.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testejä Johto- ja hallintokorvaus osiolle.
* Huom: Tuntimäärä ja -palkat taulukon alla oleva yhteenvetotaulukko jäi
testaamatta testaamisen estävän bugilöydöksen takia. Tavoite- ja
kattohinta on kuitenkin testattu, koska se hoidetaan lopuksi sivu
refreshin jälkeen.

RelNotes:
Related:

K1ll3rF0x committed:
Testaa kustannussuunnitelman osioiden yhteenvetojen luvut.

RelNotes:
Related:

K1ll3rF0x committed:
Luo TS-tyypit custom Cypress-komennoille.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa terminaalikomento cy-komennon return-tyyppi.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typo.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Testaa custom toimenkuvan lisäämistä Tuntimäärä ja -palkat taulukkoon.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Siirrä osion vahvistukseen liittyvät komponentit omaan nimiavaruuteen.

RelNotes:
Related:

K1ll3rF0x committed:
Tyylittele osion vahvistuksen UI-komponentti eri tilat huomioon ottaen.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testi osion vahvistamiselle.
* Muokattu pääyhteenvedon rakennetta testaamisen helpottamiseksi.
* Lisätty pääyhteenvedon testaamista helpottava cypress-apufunktio.

RelNotes:
Related:

K1ll3rF0x committed:
Testaa osion vahvistus tietyltä hoitovuodelta.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää TODO-kommentti.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää uusi misc/document-confirm.svg harja-ikoni.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää document-confirm ikoni osion vahvistamisen UI-komponenttiin.

RelNotes:
Related:

K1ll3rF0x committed:
Varmista, että vain aluevastaava voi vahvistaa suunnitelman osion.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää vahvistaja ja vahvistus_pvm kun osiota vahvistetaan ensimmäisen
kerran.

RelNotes:
Related:

K1ll3rF0x committed:
Toteuta ensimmäinen versio osion vahvistuksen kumoamisesta.
* Vielä on toteuttamatta vahvistamisen jälkeen tehtyjen muutosten
seuranta. Osion tila täytyisi merkata muokatuksi/likaiseksi, jotta
Kumoa vahvistus-nappula voidaan disabloida. Speksin mukaan vahvistusta
ei saa enää kumota, mikäli muutoksia taulukon lukuihin on tehtyä
vahvistuksen jälkeen
* Vahvistamisen yhteydessä ei vielä tallenneta indeksikorjattuja lukuja
tietokantaan. Kun indeksikorjatut luvut tallennetaan tietokantaan, niin
vahvistuksen kumoamisen yhteydessä pitäisi todennäköisesti tallennetut
indeksikorjatut luvut myös poistaa.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testi osion vahvistuksen kumoamiselle.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa virhe osion tilan insertoinnissa.
* Jäi kiinni cypress-testeillä. :)

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Tee tyyli- ja whitespace-korjauksia migraatioon.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää puuttuva keyword poistettavien palveluiden listaan.

RelNotes:
Related:

K1ll3rF0x committed:
Testaa uusien budjetointirajapintojen oikeustarkastukset.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää TODO-kommentteja.

RelNotes:
Related:

jpluomajoki-solita committed:
Lisää sticky-debug komponentti

 - Ei tarpeellinen, mutta hyödyllinen kustannussuunitelman kanssa.

K1ll3rF0x committed:
Lisää indeksikorjausta varten sarakkeet urakka_tavoite tauluun. Lisää
"osio"-sarake kustannusarvion lukuihin liittyviin tauluihin.

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' of github.com:finnishtransportagency/harja into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Formatoi mhu_kustannussuunnitelma

 - Formatoitu ensin cljfmt ja sitten IDEAlla.

jpluomajoki-solita committed:
Revert "Formatoi mhu_kustannussuunnitelma"

This reverts commit a7db2c9da647c76f9d11fc887cf061924aa3cdb1.

jpluomajoki-solita committed:
Paranna sticky-debug komponentttia

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Tallenna kustannussuunnitelman osio kiinteahintainen_tyo tauluun.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa migraatiosta vuosi- ja versio-sarakkeiden tietotyypit.
* Käytetään vuodelle smallint:iä ja versiolle integeriä.

RelNotes:
Related:

K1ll3rF0x committed:
Tallenna suunnitelman osio kustannusarvioitu_tyo tauluun.

RelNotes:
Related:

K1ll3rF0x committed:
Refaktoroi johto- ja hallintokorvauksia.
* Poista osion-sarake johto_ja_hallintokorvaus-taulusta.
* Refaktoroi johto- ja halintokorvausten tallentamiseen liittyvää
koodia.

RelNotes:
Related:

K1ll3rF0x committed:
Muuta osioiden app staten polun nimeä. Siirrä osioihin liittyvä apuri
mhu domainin nimiavaruuteen.

RelNotes:
Related:

K1ll3rF0x committed:
Selkeytä osioiden ja muutosten vahvistamiseen liittyviä apureita ja
tilapolkuja.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää experimentalStudio = false cypress-asetus placeholderina.
* Aseta tämä true:ksi, mikäli haluat kokeilla Cypress studiota. Ei
tarjota sitä kuitenkaan defaulttina true:na, koska kyseessä on
experimental-ominaisuus.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa bugi tilaajan rahavarausten osion vahvistamiseen liittyen.
* Täsmennä samalla eroa "Tilaajan rahavarausten" osioon ja "tilajan
varausten" gridiin liittyen.
* En uskaltanut suorilta muuttaa kaikkia "tilaajan varauksiin" viittavia
termejä "tilaajan rahavarauksiksi". Mutta, nyt ainakin osioon liittyvät
viittaukset ovat aina "tilaajan rahavarauksia".

RelNotes:
Related:

K1ll3rF0x committed:
Hankkiudu eroon hankalasta tyyppi->osio mapista. Lisää apuri, jolla
osion tunniste keyword muutetaan stringiksi tietokantaan tallennusta
vareten ja samalla tarkastetaan onko osion tunniste validi.

RelNotes:
Related:

jpluomajoki-solita committed:
Tee kattohinta-grid, laske gridiin oletusarvot

jpluomajoki-solita committed:
Lisää kattohinnan päivitys loppuihin grideihin

 - Formatoitu vanhaa koodia samalla

K1ll3rF0x committed:
Paranna kustannussuunnitelman tietomallia.
* Tallennetaan summa-riveille relevantteihin tauluihin indeksikorjatut
summat erillisiin sarakkeisiin.
* Poistettu indeksikorjattu boolean sarake ja korvattu se
indeksikorjaus_vahvistettu aikaleimalla.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää apuri jolla muunnetaan urakan vuosi hoitovuodeksi.
* Hoitovuosi on urakan hoitovuosien järjestysluku (1-5).

RelNotes:
Related:

K1ll3rF0x committed:
Lisää apurifunktiot indeksikorjaamattomien rivien hakuun tauluista.

RelNotes:
Related:

K1ll3rF0x committed:
Hahmottele indeksikorjattavien rivien hakua
budjettisuunnittelu-palveluun.

RelNotes:
Related:

jpluomajoki-solita committed:
Formatoi euro-optilla euron sijaan

jpluomajoki-solita committed:
Lisää indeksikorjatun kattohinnan laskeminen gridiin

jpluomajoki-solita committed:
Korjaa kattohinnan päivitys rahavarauksien osalta

K1ll3rF0x committed:
Lisää kustannussuunnitelman tauluihin vahvistaja-sarake.

RelNotes:
Related:

K1ll3rF0x committed:
Tee pvm-apuri, jolla voidaan muuntaa mhu hoitovuoden nro aikaväliksi.

RelNotes:
Related:

K1ll3rF0x committed:
Toteuta logiikka indeksikorjattujen summien vahvistamiselle.
* Kustannussuunnitelman osion vahvistamisen yhteydessä käydään läpi
hoitovuoden tallennetut rivit suunnitelman osiolle relevanteista
tauluista ja vahvistetaan sellaiset rivit, joita ei ole vielä
vahvistettu. Vahvistettavat rivit ovat aina versiota 0. Jälkeenpäin
tehtyjä muutoksia (eli rivit versio > 0) ei vahvisteta lainkaan, eikä
niihin lasketa indeksikorjauksia.

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' of github.com:finnishtransportagency/harja into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Korjaa typo funktion nimessä.

RelNotes:
Related:

K1ll3rF0x committed:
Tee pvm-apuri, jolla voidaan palauttaa mhu hoitovuosi-nro päivämäärälle.

RelNotes:
Related:

K1ll3rF0x committed:
[WIP] Laske ja tallenna kustannussuunnitelman indeksikorjaukset
tauluihin summia tallentaessa.

RelNotes:
Related:

K1ll3rF0x committed:
Tarkasta, että tuntipalkka on määritelty ennen indeksikorjausta.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa bugi, joka esti kustannussuunnitelman tietojen päivittymisen.
* Muutostenseurannan keskeneräinen koodi sisälsi symboleja, jotka
samannimisinä aiheuttivat ongelmia uusien summa-arvojen poimimisessa
tallennettavaksi. Nyt on tehty selvempi ero uusien ja vanhojen arvojen
välillä symbolien nimeämisessä, jotta näin ei pääse vahingossa käymään.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa bugi, joka esti indeksikorjauksen laskemisen viimeisen
hoitovuoden viimeiselle kuukaudelle.

* Lisätty uusi pvm-apuri luo-pvm-dec-kk, joka dekrementoi kk-luvun
automaattisesti. Tätä voi käyttää normaalin luo-pvm apurin rinnalla.
Koodissa näkyy hyvin paljon toistoa, jossa kuukausi dekrementoidaan
manuaalisesti ennen luo-pvm funktiolle syöttöä ja tämä voi usein myös
unohtua koodista tai lähteä vahinkossa pesuveden mukana -> fragiilia
koodia.

RelNotes:
Related:

K1ll3rF0x committed:
Kumoa vahvistus suunnitelman osion riveistä osion vahvistuksen
kumoamisen yhteydessä.
* Käy läpi osioon liittyvien taulujen rivit urakan hoitovuodelta samaan
tyyliin kuin vahvistamisen yhteydessä. Vahvistuksen kumoamisen
yhteydessä tiedot vahvistajasta ja indeksikorjaus_vahvistettu-aikaleima
poistetaan.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa jeesql-funktioiden nimet budjettisuunnittelu-palvelussa.

RelNotes:
Related:

K1ll3rF0x committed:
Toteuta toiminto, jolla tavoite-ja kattohinta osio kumotaan myös mikäli
siihen liittyvä suunnitelman osio on kumottu.

RelNotes:
Related:

K1ll3rF0x committed:
Estä tavoite-ja kattohinta osion vahvistaminen UI:ssa, jos siihen
liittyviä osioita ei ole vielä vahvistettu.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Laske indeksikorjattu kattohinta uuteen gridiin

K1ll3rF0x committed:
Urakan indeksikertoimia haettaessa palautetaan nil mikäli hoitovuoden
indeksikerrointa ei ole saatavilla.
* Näin estetään indeksikorjattujen lukujen laskeminen tietokantaan
väärillä kertoimilla.

RelNotes:
Related:

jpluomajoki-solita committed:
Tyylittely kattohintataulukolle

jpluomajoki-solita committed:
Otsikko kattohinnalle

jpluomajoki-solita committed:
Kattohinnan taulukon oletusarvoksi 0

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' of github.com:finnishtransportagency/harja into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Näytä manuaalinen kattohinnan asetus oiken

 - Näytetään manuaalisen kattohinnan asetuksen taulukko vain urakoillle,
   jotka ovat alkaneet 2019 tai 2020.

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Alustava manuaalisen kattohinnan tallennus

K1ll3rF0x committed:
Ota huomioon tyhjät indeksit ja arvot kustannussuunnitelman
cypress-testien apufunktioissa.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa kustannussuunnitelman cypress-testit (paitsi kattohinnan testi)
* Kattohinnan manuaalinen syöttö on nyt näkyvissä testiurakan
suunnitelussa, joten vanha testi ei enää toimi.
* Täytyy kehittää keino, jolla voidaan testata sekä manuaalista syöttöä,
että vanhanmallista kattohinnan yhteenvedon testaamista samalla
testiurakalla. Joudutaan ehkä vaihtamaan urakkaa testien välissä.

RelNotes:
Related:

jpluomajoki-solita committed:
Kommentoi kattohinnan laskemisen testi

K1ll3rF0x committed:
Hae indeksikorjaukset tietokannasta appin tilaaa (paitsi tavoite- ja
kattohinta).

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' of github.com:finnishtransportagency/harja into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Lue alusta määrätaulukkojen data tietokannasta. Näytä erillishankintojen
osioiden indeksikorjaukset tietokannasta osion yhteenvedoissa.

RelNotes:
Related:

K1ll3rF0x committed:
Kokeile päivittää kustannushintojen tila siihen liittyvän grid-kentän
päivittämisen jälkeen. (Karvalakki)
 * Tämä on todella hidas. Pitäisi olla rajapinta, joka palauttaa vain
tiettyyn gridiin liittyvät muutokset kentän päivityksen jälkeen.
Grid puolella datalle tehdään raskasta operointia ja sitä pilkotaan
useaan paikkaan. Rajapinnat pitäisi suunnittella uusiksi.
* Näyttää siltä, että joudumme elämään pidemmän aikaa niin, että
UI-yrittää mallintaa omilla laskutoimituksillaan sitä mitä tietokantaan
on tallennettu.
* Tai, sitten kylmästi käytetään UI-puolen indeksikorjausten laskentaa
ja lähetetään payloadissa indeksikorjatut luvut tietokantaan
tallennettavaksi ja luotetaan siihen, että UI:sta lähtee tietokantaan
aina oikeat indeksikorjatut luvut.

RelNotes:
Related:

K1ll3rF0x committed:
Mahdollista custom result-eventtien käyttö kustannusarvioitujen töiden
tallentamiseen. Käytä yksinkertaisempaa eventtiä Erillishankintojen
määrägridin tallentamisen jälkeiseen tilanhallintaan.

RelNotes:
Related:

K1ll3rF0x committed:
Yksinkertaista gridien tilan alustusta. Siirrä tuck-eventin sisällä
määritellyt apufunkiot ulos event handlerista.

RelNotes:
Related:

jpluomajoki-solita committed:
Kattohinnan manuaalisen gridin cypress-testit

K1ll3rF0x committed:
Päivitä toimistokulujen ja hoidonjohtopalkkion gridien tilat backendin
datalla tallentamisen yhteydessä.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa tilaajan rahavaukset osion tallennusongelma.
* TallennaKustannusarvioituOnnistui ei palauttanu app-tilaa, joten se
kaatoi koko web-sovelluksen.

RelNotes:
Related:

K1ll3rF0x committed:
Tee testaaRivienArvot cypress-funktiosta automaattisesti retryttävä.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testien lokitusta.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

K1ll3rF0x committed:
Vaihda käytöstä poistetun ikonin tilalle uusi harjan ikonipankista.

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

K1ll3rF0x committed:
Korjaa MHU Ylläpidon Toimenpiteet rahavaraukset-gridin "Muut
rahavaraukset" tallentaminen.
* Tallennettava-asia->osio-kw mapista puuttui :rahavaraus-lupaukseen-1
-> :hankintakustannukset

RelNotes:
Related:

jpluomajoki-solita committed:
Hae budjettitavoite kustannussuunitelmassa

 - Käytetään manuaalisen kattuhinnan taulukossa.

jpluomajoki-solita committed:
Kommentoi rikki menneitä testejä toistaiseksi

jpluomajoki-solita committed:
Korjaa kattohinnan haku, jos sitä ei ole kannassa

jpluomajoki-solita committed:
Korjaa muutama kuun vaihteessa hajonnut cypress-testi

jpluomajoki-solita committed:
Tee testiurakka 2019-2020 alkaneille MH-urakoille

jpluomajoki-solita committed:
Merge branch 'develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Poista tilaajan rahavaraukset osion vahvistus pois käytöstä. Poista
osion status ja indeksikorjattu summa tilaajan rahavaraukset-osion
yhteenvedosta.

RelNotes:
Related:

K1ll3rF0x committed:
Estä indeksikorjauksen laskeminen "tilaajan rahavaraukset"
kustannusarvioidulle työlle.

RelNotes:
Related:

K1ll3rF0x committed:
Nimeä käyttöliittymässä "Tilaajan rahavaraukset" -> "Tavoitehinnan
ulkopuoliset rahavaraukset".

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

K1ll3rF0x committed:
Korjaa tilaajan rahavaraukset-osion e2e-testit.

RelNotes:
Related:

K1ll3rF0x committed:
Muuta testin nimi.

RelNotes:
Related:

jpluomajoki-solita committed:
Lisää validointi manuaalisen kattohinnan gridille

jpluomajoki-solita committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Korjaa nullpointer-erhe indeksikorjauksess, kun tallennettava summa on
nil.

RelNotes:
Related:

jpluomajoki-solita committed:
Siirrä urakan alustusfunktio

K1ll3rF0x committed:
Piilota datafrisk defaulttina. Näytä hotkeyllä ctrl+b tai meta+b.

RelNotes:
Related:

K1ll3rF0x committed:
Palauta aiempi keskeneräinen vahvistudialogi toimintakuntoon.

RelNotes:
Related:

K1ll3rF0x committed:
Näytä varoitus, jos vahvistettuun osioon yritetään tehdä muutoksia.
* Näytetään väliaikainen varoitusdialogi.
* Muutosten tekemistä grid input-kenttiin ei ole estetty tässä, mutta
muutosten tallentaminen on estetty vahvistettuihin osioihin.

RelNotes:
Related:

K1ll3rF0x committed:
Disabloi kaikki osion gridien input-kentät mikäli osio on vahvistettu.
* Tämä on väliaikainen ratkaisu, kunnes muutosten seuranta saadaan
toteutettua.

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

jpluomajoki-solita committed:
Paranna kattohinnan gridin validointi

 - Ulkoisten muuttujien käyttö validoinnissa vaati pientä kikkailua

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

jpluomajoki-solita committed:
Formatoi virheviesti paremmin ja kommentoi koodia

K1ll3rF0x commented on src/cljs/harja/views/urakka/suunnittelu/kustannussuunnitelma/tavoite_ja_kattohinta_osio.cljs:
```diff
@@ -37,43 +37,51 @@
-                                            (range 1 6)))))})
-       tiedot/kustannussuunnitelma-kattohinta]]
-     [yleiset/ajax-loader]))
+  [_ tavoitehinnat _]
```
Mihin tarvitaan ylimääräistä atomia ja tällainen rakenne? 
Eikös esim. tämä olisi riittävä:

```cljs

(defn- manuaalinen-kattohinta-grid
  [e! tavoitehinnat kantahaku-valmis?]
   (fn [e! tavoitehinnat kantahaku-valmis?]
     [:h5 "Kattohinta"]
     (if kantahaku-valmis?
       [:div
        {:on-blur #(e! (t/->TallennaJaPaivitaTavoiteSekaKattohinta))}
        [v-grid/muokkaus-grid
         {:otsikko "Kattohinta"
          :luokat ["kattohinta-grid"]
          :data-cy "manuaalinen-kattohinta-grid"
          :piilota-toiminnot? true
          :muokkauspaneeli? false
          :ulkoinen-validointi? true
          :virheet t/kattohinta-virheet
          :virheet-dataan? true
          :luomisen-jalkeen #(e! (t/->ValidoiKattohintaGrid tavoitehinnat))
          :muutos #(do (e! (t/->PaivitaKattohintaGrid %))
                       (e! (t/->ValidoiKattohintaGrid tavoitehinnat)))
          :valiotsikot {1 (v-grid/otsikko "Indeksikorjatut")}
          :disabloi-rivi? #(not= :kattohinta (:rivi %))
          :sisalto-kun-rivi-disabloitu #(fmt/euro-opt ((:nimi %) (:rivi %)))}
         (merge
           (mapv (fn [hoitovuosi-numero]
                   {:otsikko (str hoitovuosi-numero ".hoitovuosi")
                    :nimi (keyword (str "kattohinta-vuosi-" hoitovuosi-numero))
                    :fmt fmt/euro-opt
                    :tyyppi :positiivinen-numero})
             (range 1 6))
           {:otsikko "Yhteensä"
            :nimi :yhteensa
            :tyyppi :positiivinen-numero
            :muokattava? (constantly false)
            :disabled? true
            :fmt fmt/euro-opt
            :hae #(apply + (vals
                             (select-keys % t/kattohinta-grid-avaimet)))})
         t/kattohinta-grid]]
       [yleiset/ajax-loader])))

jpluomajoki-solita replied:
Uusin versio toimii ainakin täydellisesti :+1:

K1ll3rF0x commented on src/cljs/harja/tiedot/urakka/suunnittelu/mhu_kustannussuunnitelma.cljs:
```diff
@@ -1766,20 +1778,20 @@
            :yhteensa {:nimi "Yhteensä"}
            :kuukausitasolla? false})
-        (assoc-in [:kattohinta 0]
+        (assoc-in [:kattohinta :grid 0]
```
Nämä voisi generoida (range 1 6) avulla ja välttää käsin kirjoittamista.

Kuulin, että periaatteessa voisi olla urakoita joskus tulevaisuudess, joilla on esim. 7 hoitovuotta.
Siinä tapauksessa pitää joskus etsiä koodista kaikki paikat missä on generoitu avaimia rangejen avulla, niin tuollaiset (range 1 6) viittaukset löytyisivät kohtalaisen helposti lähdekoodista.

Pitäisi joskus muutenkin refaktoroida koodi sellaiseksi, että se tukee hoitovuosien määrän säätöä dynaamisemmin... Mutta, se on sitten isompi jumppa osa isompaa refaktoroinurakkaa.

K1ll3rF0x commented on src/cljs/harja/tiedot/urakka/suunnittelu/mhu_kustannussuunnitelma.cljs:
```diff
@@ -189,6 +189,17 @@
 (def manuaalisen-kattohinnan-syoton-vuodet
   [2019 2020])
 
+(def kattohinta-grid-avaimet
```
Tämän voi alustaa (range 1 6) avulla, mikäli haluaa säästää tämän mapin ja helpottaa silllä select-keys käyttöä toisaalla.

K1ll3rF0x committed:
Ota huomioon muuttunut skeema muokkaus gridin ohjaus-fn:ssa.
* Korjaus mahdollistaa mm. dynaamiset validointisäännöt, joissa on
vaihtuvat arvovälit.

RelNotes:
Related:

K1ll3rF0x committed:
Palauta käyttöön validointisääntö :manuaalinen-kattohinta ja käytä
ohjauskahvaa manuaalisen kattohinnan gridin validoinnin apuna.
* Poistettu aiempi validoinnin hack.
* Otettu huomioon uusi paranneltu validoinnin virheviesti takaisin
vanhaan validointiin siirtyessä.

RelNotes:
Related:

jpluomajoki-solita committed:
Alusta kattohinta rangella

jpluomajoki-solita committed:
korjaa kattohinnan avainten map

K1ll3rF0x committed:
Pyöristä indeksikorjattu summa kuuden desimaalin tarkkuuteen
tietokantaan.
* Indeksikorjatun summan tarkkuus tulisi olla 4-6 desimaalia
tietokannassa, jotta summan tarkkuus on riittävä laskentaan muualla.
Lopputulos voidaan pyöristää pienempään tarkkuuteen käyttöliittymässä.

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'kustannussuunnitelman-osittaminen' into lue-maarataulukkojen-indeksikorjaukset-tietokannasta

K1ll3rF0x committed:
Korjaa bugi hoidonjohtopalkkio-osion summien alustamisessa.
* Sivun latauksen yhteydessä laitettiin vanhingossa indeksikorjatut
summat indeksikorjaamattomien summien tilalle hoidonjohtopalkkio-osion
yhteenvedon app-tilaan.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää TODO-kommentteja liittyen indeksikorjausten hakemiseen
tietokannasta.
* Vielä pitää hakea indeksikorjaukset Hankintakustannukset osion ja
Johto- ja hallintokorvaus osion grideille (paitsi
määrätaulukko-tyyppisille grideille).
* Tätä on suunniteltu TODO-kommenttien avulla. Tehdään osio kerrallaan,
jotta voidaan testata muutoksia lopulta osio kerrallaan.

RelNotes:
Related:

K1ll3rF0x committed:
Paranna gridien "johdettujen arvojen" varoitusviestejä.

RelNotes:
Related:

K1ll3rF0x committed:
Poista käytöstä gridien johdettujen arvojen varoitusviestit, koska ne
antoivat turhaa tietoa. Ne voi enabloida uudestaan mikäli epäilee jonkin
bugin liittyvän johdettuihin arvoihin.
* Johdettuihin arvoihin liittyvät arvot alustetaan gridien
datakäsittelijöissä, jonka takia johdettujen arvojen pituus arvoihin
verrattuna on aina eri ja kustannussuunnitelma välilehden latautuessa
tulee lokiin läjäpäin turhia warning-tyyppisiä lokituksia.

RelNotes:
Related:

jpluomajoki-solita committed:
Manuaalisen kattohinnan cypress-testit

jpluomajoki-solita committed:
Älä tallenna virheellistä kattohintaa

jpluomajoki-solita committed:
Merge branch 'develop' into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Nimeä migraatiotiedosto oikeaksi

K1ll3rF0x committed:
Kirjoita backend-testit indeksikorjauksien laskemiselle.
* Testit kirjoitettu kiinteähintaisille töille, kustannusarvioiduille
töille ja johto- ja hallintkorvauksille.

RelNotes:
Related:

jpluomajoki-solita committed:
Tee backend-validaatio kattohinnalle

jpluomajoki-solita committed:
Älä lähetä virheellistä kattohintaa

jpluomajoki-solita committed:
Älä lähetä tyhjää kattohintaa

K1ll3rF0x committed:
Testaa budjettitavoitteen tallennuksen indeksikorjauksen oikeellisuus.
* Testattu lasketut ja tietokantaan tallennetut tavoitehinnan ja
kattohinnan indeksikorjatut summat.

RelNotes:
Related:

jpluomajoki-solita committed:
Korjaa kattohinnan bad case-skenaario

K1ll3rF0x committed:
Testaa osioihin liittyvien tietokantataulujen vahvistetut rivit.
* Tallennetaan backend-testeissä kustannussuunnitelman osioihin
liittyviin tietokantatauluihin arvoja, jotka vahvistetaan kutsumalla
osion vahvistamiseen rakennettua rajapintaa, ja lopuksi tarkastetaan
onnistuiko osioihin liittyvien rivien vahvistus.

RelNotes:
Related:

K1ll3rF0x committed:
Poista debug-print.

RelNotes:
Related:

K1ll3rF0x committed:
Varmista, että osion vahvistuksen testissä tarkastetaan, että
vahvistettuja rivejä on vain yhden hoitovuoden verran.
* Testiä pitää vielä parantaa siten, että tarkastetaan vahvistettuja
rivejä olevan vahvistetun hoitovuosi-nro:n hoitokauden aikavälin
sisällä. Eli, varmistutaan siitä, että ei vahingossa vahvisteta väärän
hoitovuoden rivejä. Tällainen voi tapahtua esim. jos pvm-apufunktioissa
on jokin bugi, mikä rikkoo hoitokauden alkupvm ja loppupvm laskennan.

RelNotes:
Related:

K1ll3rF0x committed:
Varmista, että osion vahvistuksen backend-testissä vahvistettujen rivien
vuosi/kuukausi tai hoitokausi arvot osuvat samalle hoitovuodelle, joka
vahvistettiin.

RelNotes:
Related:

K1ll3rF0x committed:
Poista lokituksia, poista TODO-kommentti.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Salli kattohinnan olla nolla, mikäli tavoitehintakin on nolla.
* Korjaa mhu_kustannussuunnitelma.js cypress-testit.

RelNotes:
Related:

jpluomajoki-solita committed:
Korjaa urakanhakutesti

K1ll3rF0x committed:
Korjaa tavoite-ja kattohinnan backend-testi.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa Suunnitellut Hankinnat gridin "Maksetaan"-valikon virhe.
* Korjaa VHAR-5423

RelNotes:
Related:

jpluomajoki-solita committed:
Lisää apuri testidatan generointiin

jpluomajoki-solita committed:
Paranna indeksikertoimen pyöristelyä

 - Aiemmin perusluku muutettiin floatiksi, minkä tarkkuus ei riittänyt
   indeksikertoimen laskuun oikein.

K1ll3rF0x committed:
Korjaa indeksikertoimen haun testi.
* Indeksikeroimen laskentaa parannettiin ja lopputulos tarkentui. Tämä
vaikutti indeksikertoimen desimaalien arvoihin ja rikkoi
indeksikertoimen hakuun liittyvän testin.

RelNotes:
Related:

jpluomajoki-solita committed:
Lisää indeksikorjausten laskentaa testidataan

 - Sisältää indeksikorjausten laskennan suunniteltuihin töihin
 - Indeksikorjausten laskenta on väärin. Laskee kalenterivuodella eikä
   hoitokaudella

K1ll3rF0x committed:
Poista "osio"-sarake kiinteahintaiset_tyot taulusta.
* Koska kiinteähintaisia töitä tulee tällä hetkellä ainoastaan
kustannussuunnitelman Hankintakustannukset-osiosta, ei ole tarpeen
tallentaa tietoa mistä osiosta rivi on kiinteahintaiset_tyot-tauluun
tullut.

RelNotes:
Related:

jpluomajoki-solita committed:
Korjaa testidatan indeksikorjaukset

jpluomajoki-solita committed:
korjaa yksittäisiä virheitä testidatassa

K1ll3rF0x committed:
Korjaa Kustannussuunnitelman lomakkeiden rikkoutuminen urakan vaihdon
yhteydessä.
* Jos urakkaa vaihtoi ylävalikosta kustannussuunnitelma-välilehden
ollessa auki, jäi kustannussuunnitelman gridit siivoamatta. Tämä johtui
suunnitelmien yhteisen tilan default-tilaan resetoinnista siinä
yhteydessä, kun urakkaa vaihdetaan. Tämä tapahtui ennen kuin
kustannussuunnitelman UI-komponentti ehti on-unmount lifecycleen ja
rikkoi gridien pois siivoamisen.

* Korjaa VHAR-4909

RelNotes:
Related:

K1ll3rF0x committed:
Piilota Hankintakustannukset (ennen urakkaa)-rivi, mikäli valittu
hoitovuosi on muu kuin 1.

Korjaa: VHAR-3127

RelNotes:
Related:

K1ll3rF0x committed:
Lisää cypress-testit "Hankintakustannukset (ennen
urakkaa)"-toimenkuvalle.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa virhe "Hankintavastaava (ennen urakkaa)" yhteevetosummassa
* Tämän toimenkuvan tunnit kerrottiin tarpeettomasti "osa-kuukaudesta"
arvolla. Tällöin kävi niin, että viimeisen kuukauden tunnit kerrottiin
0.5-kertoimella, jolloin loppusumman laskenta meni pieleen. Loppusumma
lasketaan tyyliin tunnit * osa-kuukaudesta.
Eli, jos tunnit = 1 ja osa-kuukaudesta on 0.5, niin virheen takia tunnit
kerrottiin ensin 0.5:llä, jolloin tunnit = 0.5. Ja loppusumman
laskennassa tunnit * osa-kuukaudesta = 0.5 * 0.5 = 0.25.
* Tietokantaan tunnit tallennettiin kuitenkin oikein kokonaisina
johto-ja hallintakorvauksien tauluun. Tallennus tapahtui ennen
yhteenvetoa varten tehtyä mutaatiota, joten ongelma näkyi vain
käyttöliittymässä.
* Huomattavaa on kuitenkin, että tavoite- ja kattohinta lasketaan ja
päivitetään tietokantaan käyttöliittymässä näkyvien yhteenvetosummien
perusteella, jolloin tämä bugi on voinut aiheuttaa virhettä tavoite- ja
kattohintaan. Eli, ongelma ei välttämättä ole ollut pelkästään
visuaalinen.

Korjaa: VHAR-5257

RelNotes:
Related:

K1ll3rF0x committed:
Nimeä migraatio uudelleen.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Korjaa kustannussuunnitelma gridin z-index ongelmat kk-valinnan
dropdownien kanssa.
* Osa kuukausivalinta-drowdownista jäi gridin alle piiloon
custom-toimenkuva riveillä.

RelNotes:
Related:

K1ll3rF0x committed:
Varmista, että gridin :pudotusvalikko-solutyypin elementin id on
uniikki.

RelNotes:
Related:

topipoh committed:
Lisää alustava migraatio kustannussuunnitelman olemassa olevalle datalle

K1ll3rF0x committed:
Paranna kustannussuunnitelman "poisto-dialogia" niin, että suurempi
määrä tietoa mahtuu näkyville kerralla ja että toimintonappulat tulevat
esille.

Korjaa: VHAR-5511

RelNotes:
Related:

K1ll3rF0x committed:
Poista gridin pudotusvalikko-solun zindex kikkailut.

RelNotes:
Related:

K1ll3rF0x committed:
Siirrä maksukausi<->kuukausi mappaukset apufunktioiksi mhu-domainiin.

RelNotes:
Related:

jpluomajoki-solita committed:
Formatoi migraatio IDEA:n oletusten mukaan

K1ll3rF0x committed:
Korjaa kuukausitaulukon rikkoutuminen custom toimenkuvan kuukausia
vaihtaessa.

Korjaa: VHAR-5445

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa custom toimenkuvan kuukausien vaihto.
* Nyt tietokantaan päivittyy custom-toimenkuvan kuukaudet, mikäli sen
vaihtaa käyttöliittymän kautta.
* Yhteenvedon summalaskennassa havaittu virhe korjattu. Summa
hoitovuodelle oli välillä nolla, koska tuntipalkka otettiin summaan aina
ensimmäisestä hoitovuoden kuukaudesta. Riippuen kuukausivalinnasta
ensimmäisellä kuukaudella ei välttämättä ollut ollenkaan
tuntipalkkatietoja.

Korjaa: VHAR-2723, VHAR-2724

RelNotes:
Related:

K1ll3rF0x committed:
Paivita custom-toimenkuvalle vain maksukauteen liittyvät kuukaudet, kun
tunteja tai tuntipalkkaa muokataan.
* Aiemmin päivitettiin kaikki 12 kuukautta, vaikka olisi valittu vain 5
tai 7 kuukautta/vuosi. Tämä johti siihen, että tietokantaan
tallennettiin hoitovuoden kustannukset aina 12 kuukaudelle riippumatta
kuukausivalinnasta.

Korjaa: VHAR-5522

RelNotes:
Related:

K1ll3rF0x committed:
Jos "tunnit" input-kenttää muokataan, niin päivitä samalla varmuuden
vuoksi tuntipalkka applikaation tilaan.

Korjaa: VHAR-5520

RelNotes:
Related:

jpluomajoki-solita committed:
Täytä osio-sarake vanhoille kustannuksille

K1ll3rF0x committed:
Korjaa datan poistaminen tietokannasta, kun custom-toimenkuvan nimi
pyyhitään.
* Nimi jätetään kuitenkin vielä toimenkuvat-tauluun tietokantaan.
Kustannukset poistetaan johto- ja hallintokorvaus taulusta.

Korjaa: VHAR-5521

RelNotes:
Related:

topipoh committed:
Muuta otsikko: Toimenpiteen rahavaraukset (lasketaan kattohintaan)

K1ll3rF0x committed:
Korjaa kustannussuunnitelman "poisto-modalin" virheellisesti näytetyt
kuukausisummat.
* Riveillä näkyi vain tunnit muka kuukausisummina johto- ja
hallintokorvaus-osiossa.
* Paranna rivien white-spacea.

RelNotes:
Related:

topipoh committed:
Muuta pääotsikko "Kustannussuunnitelma" h3->h1

topipoh committed:
Muuta otsikko "Erillishankinnat" h3->h2

topipoh committed:
Muuta otsikko "Johto- ja hallintokorvaus" h3->h2

topipoh committed:
Muuta otsikko "Hoidonjohtopalkkio" h3->h2

topipoh committed:
Muuta otsikko "Tavoite- ja kattohinta" h3->h2

K1ll3rF0x committed:
Korjaa ongelma tavoitehinnan ja johto- ja hallintokorjauksen
yhteenvetojen päivityksessä, kun "Tuntimäärät ja palkat" arvoja
syötetään ensimmäistä kertaa.
* Ensimmäistä kertaa arvoja syöttäessä yhteenvedot ja tavoitehinta ei
päivittynyt ollenkaan. Tämä johtui siitä, että osa-kuukaudesta
default-arvoa ei oltu määritelty käyttöliittymän puolella ollenkaan,
vaan se tuli ainoastaan tietokannasta.

Korjaa: VHAR-3505, VHAR-5226

RelNotes:
Related:

K1ll3rF0x committed:
Seuraa vakio, ja custom-toimenkuvarivien muutoksia ja aseta
default-arvot tarvittaessa.
* Korjaa default arvojen puutteiden ongelmia, mikäli muokataan rivin
kuukaudet-alivalikkoa.

Korjaa: VHAR-3505, VHAR-5520

RelNotes:
Related:

topipoh committed:
Muuta otsikossa: "lasketaan kattohintaan" -> "lasketaan tavoitehintaan"

"Toimenpiteen rahavaraukset (lasketaan kattohintaan)"
->
"Toimenpiteen rahavaraukset (lasketaan tavoitehintaan)"

topipoh committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into kustannussuunnitelman-osittaminen

K1ll3rF0x committed:
Tarkasta, että toimenkuvalle on olemassa domain-tilassa arvoja valitulle
hoitovuodelle ennen default-arvojen asettamista.
* Hankintakustannukset (ennen urakkaa)-toimenkuvalle on arvoja vain
ensimmäiselle hoitovuodelle, joten se aiheutti hallitsemattoman virheen
hoitovuotta vaihtaessa ja domain-tilan seurannan triggeröityessä.

RelNotes:
Related:

jpluomajoki-solita committed:
Tarkenna hankintakustannusten osion migraatioita

jpluomajoki-solita committed:
Migratoi myös ennen 2019 alkaneet MH- ja HJ-urakat

jpluomajoki-solita committed:
Hoitokauden valinta manuaaliseen kattohintaan

jpluomajoki-solita committed:
Älä lähetä kattohintaa, jos se on alle tavoitteen

jpluomajoki-solita committed:
Estä tavoite- ja kattohinnan varmistus, jos kattohinnassa virheitä.

K1ll3rF0x committed:
Korjaa tietokannan testidata Pellon ja Kemin urakalle Hankintavastaava
(ennen urakkaa) osalta.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa budjettisuunnittelu-indeksit-vastaus spec.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa VHAR-5530

RelNotes:
Related:

topipoh committed:
Disabloi kustannussuunnitelman kentät, jos ei ole kirjoitusoikeutta.

topipoh committed:
Älä lähetä tallennuspyyntöä, jos käyttäjällä ei ole kirjoitusoikeutta.

topipoh committed:
Nimeä laheta-ja-odota-vastaus -> tallenna-ja-odota-vastaus

Funktio käyttää tallennusjonoa ja tallennuskanavaa, joten sitä on
tarkoitus käyttää vain tallennusoperaatioihin.

topipoh committed:
Merge remote-tracking branch 'origin/kustannussuunnitelman-osittaminen' into kustannussuunnitelman-osittaminen

# Conflicts:
#	dev-resources/less/pages/kustannussuunnitelma.less

jpluomajoki-solita committed:
Korjaa manuaalisen kattohinnan cypress-testit

K1ll3rF0x committed:
Lisää maksuerat.sql testidataan puuttuva Kittilän urakka.
* Korjaa ongelmat testiympäristön kustannusten seurannassa Kittilän
urakan kohdalla.

RelNotes:
Related:

K1ll3rF0x committed:
Nimeä migraatiot uudelleen.

RelNotes:
Related:

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen

jpluomajoki-solita committed:
Korjaa VHAR-5533

 - Korjaa määrämitattavien suunniteltujen hankintojen poisto.

topipoh committed:
Ei laiteta suunnitellut ja laskutus -mappeja settiin (voi olla samat)

K1ll3rF0x committed:
Merge remote-tracking branch 'origin/develop' into kustannussuunnitelman-osittaminen
